File: flink-datastream-1.7.1/src/main/scala/scalalang/function/TemperatureFilter.scala
package scalalang.function

import org.apache.flink.api.common.functions.FilterFunction

import scalalang.util.SensorReading

class TemperatureFilter(threshold: Long) extends FilterFunction[SensorReading] {
  override def filter(t: SensorReading): Boolean = t.temperature >= threshold
}

File: flink-datastream-1.7.1/src/main/java/function/TemperatureFilter.java
package function;

import org.apache.flink.api.common.functions.FilterFunction;
import util.SensorReading;

public class TemperatureFilter implements FilterFunction<SensorReading> {
    private final double threshold;
    public TemperatureFilter(double threshold) {
        this.threshold = threshold;
    }
    @Override
    public boolean filter(SensorReading sensorReading) throws Exception {
        return sensorReading.temperature >= threshold;
    }
}
